### PR TITLE
PRM-3813 Handling Redirect to Https When Loading The Widget.

### PR DESCRIPTION
**Pull request by georgehlim**
* When widgets load through redirect (to https), the resize event is ignored because the iframe src is in http and the contents is in https.
* Updated the hasOrigin method to strip away any protocol.
* If the url matches the iframe origin without the protocol, then hasOrigin will still return true.
* In this case the resize event will not be ignored and the widget will show even during a redirect.

This PR is based on this one in web: https://github.com/CrowdTwist/web/pull/2628

Because of the above PR, when widgets redirect to HTTPS, the widget doesn't resize and still appears hidden.  This is because the `hasOrigin` method below compares the url of the iframe to the widget origin.  This is the full url including the protocol.  Since the protocols are different `hasOrigin` returns false and therefore doesn't resize the widget.

To fix this, I've updated the function to strip out any protocol and only compare to see if the widget origin exist in the iframe origin.

- [x] @bertrandvidal 
- [ ] @cokuyan 
- [x] @c-wright 
- [ ] @dkarten 
- [ ] @amod-j 
- [ ] @jtaoct 
- [ ] @matthewsheahan

### CODE REVIEW

**Comment by c-wright** [comment]
the path for the sdk would be `resources/widgets/javascripts/widgets-sdk.0.2.9.js` in web

File: src/widget.coffee [hunk 1]
```diff
@@ -239,7 +238,8 @@
     hasOrigin: (origin) ->
       hasOrigin = false
       for widget in @_widgets
-        hasOrigin = true if origin.indexOf(widget.origin) isnt -1
+        originWithoutProtocol = origin.replace /(^\w+:|^)\/\//, ''
```
Comment (cokuyan) [inline src/widget.coffee]: This also matches and replaces anything that just starts with `"//"`, is that intentional?

Comment (georgehlim) [inline src/widget.coffee]: yup.

**Comment by georgehlim** [comment]
@cokuyan responded.

**Pull request merged**